pkg/sriov/sriov.go

```diff
@@ -234,6 +237,8 @@ func (s *sriovManager) ApplyVFConfig(conf *sriovtypes.NetConf) error {
 		if err = s.nLink.LinkSetVfHardwareAddr(pfLink, conf.VFID, hwaddr); err != nil {
 			return fmt.Errorf("failed to set MAC address to %s: %v", hwaddr, err)
 		}
+	} else {
+		conf.MAC = pfLink.Attrs().Vfs[conf.VFID].Mac.String()
```

Comment:
`pfLink.Attrs().Vfs[conf.VFID].Mac.String()` may be zero MAC

Reply:
also this is mixing desired state with actual state, personally not a fan.

In addition it will needlessly trigger setup of MAC address in SetupVF (L85).